### PR TITLE
require valid timezone string

### PR DESCRIPTION
**Pull request by ali-graham**
The `timezone` property can now only be changed by a string accepted by `ActiveSupport::TimeZone`, and `time_zone` is an accessor for the related TimeZone object.

Fixes part of #791.

### CODE REVIEW

**Comment by ali-graham** [comment]
I've normalised master to have similar behaviour, although that also validates properly: https://github.com/flapjack/flapjack/commit/b1815cc6985f8b670522609ae5dce410e159a3cb.

File: features/steps/events_steps.rb [hunk 1]
```diff
@@ -362,7 +362,7 @@ def icecube_schedule_to_time_restriction(sched, time_zone)
 
 Given /^user (\S+) has the following notification rules:$/ do |contact_id, rules|
   contact = Flapjack::Data::Contact.find_by_id(contact_id, :redis => @redis)
-  timezone = contact.timezone
+  time_zone = contact.time_zone
```
Comment (jessereynolds) [inline features/steps/events_steps.rb]: why are we s/timezone/time_zone/ ? the attribute is 'timezone' in the api http://flapjack.io/docs/1.0/jsonapi/#contacts so it'd be nice to be consistent unless there's a good reason

Comment (jessereynolds) [inline features/steps/events_steps.rb]: timezone is a String, time_zone is a ActiveSupport::TimeZone ?

Comment (jessereynolds) [inline features/steps/events_steps.rb]: yep ok, ignore me

Comment (ali-graham) [inline features/steps/events_steps.rb]: Yeah, there was some confusion internally in the code prior. 'timezone' as a string should always be the external interface, and `time_zone` is used for the instantiation thereof.

**Comment by jessereynolds** [comment]
Tops

**Pull request merged**